src/Slider.js

```diff
@@ -3,17 +3,61 @@ import PropTypes from 'prop-types';
 import cx from 'classnames';
 
 class Slider extends Component {
-  componentDidMount() {
-    const { fullscreen, indicators, interval, transition } = this.props;
+  constructor() {
+    super();
+    this.initSlider = this.initSlider.bind(this);
+    this.fullscreenReset = this.fullscreenReset.bind(this);
+    this.setActiveIndex = this.setActiveIndex.bind(this);
+  }
 
-    $('.slider').slider({
-      full_width: fullscreen,
+  initSlider() {
+    const { indicators, interval, duration, height } = this.props;
+    this.instance = M.Slider.init(this._slider, {
       indicators,
       interval,
-      transition
+      duration,
+      height
     });
   }
 
+  /**
+   * If the slider was not in fullscreen, the height is set as a style attribute
+   * on the Slider element. When `.destroy()` is called, this attribute is not
+   * removed, resulting in a fullscreen displayed incorrectly.
+   */
+  fullscreenReset(was_fullscreen) {
+    if (!was_fullscreen && this.props.fullscreen) {
+      this.instance.el.removeAttribute('style');
+      this.instance.el.childNodes[0].removeAttribute('style');
+    }
+  }
+
+  setActiveIndex(activeIndex) {
+    if (this.props.indicators && activeIndex)
+      this.instance['$indicators'][activeIndex].className =
+        'indicator-item active';
+  }
+
+  componentDidMount() {
+    this.initSlider();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (!this.instance) return;
+    const active_index = this.instance.activeIndex;
+    this.instance.destroy();
+    this.fullscreenReset(prevProps.fullscreen);
+    this.initSlider();
+    // keep indicator at current index displayed
+    this.setActiveIndex(active_index);
+  }
```

Comment:
Just for documentation, could you explain in which scenario this would happen? As in, when would I update this component and want to keep the same `activeIndex`? :)

Reply:
Whenever a react component get updated props, it should keep its state. Internally in the component it is not part of the state, but as a user, I would consider it so.

Reply:
Agree, thanks for the explanation